### ISSUE TITLE
VPC peering not handled correctly (unverified)

### ISSUE THREAD

**Issue by vr00n** [duo-labs/cloudmapper#72]
I have configured my `config.json` to contain all my CIDRs yet the diagram shows my data-vpc as being disconnected from the other 2 VPCs despite there being peering connections and route tables to ensure connectivity between all 3 VPCs.

Am I missing something here?

![image](https://user-images.githubusercontent.com/4397663/37352290-91420344-26a2-11e8-9e84-6359f2652e30.png)

**Comment by 0xdabbad00** [comment]
CloudMapper looks at the Security Groups that applied to resources (ex. an EC2 instance) and determines what could connect to it. For example, in the demo the RDS instances have a Security Group called `Database` on them that allows in connections from anything with the Security Group `Internal Web`, and that Security Group has been applied to the EC2 instances named `Web1` and `Web2`.

The ELB's have a Security Group named `Public Web` and this allows in any `0.0.0.0/0` CIDR.

![cloudmapper-fig1](https://user-images.githubusercontent.com/1979334/37320113-ce7507e0-2637-11e8-8aba-f7ca7798b5b5.png)

The code should take into account the VPC peering connections.  It finds these connections here: https://github.com/duo-labs/cloudmapper/blob/a116da65b72af48398541eb2e37c768c76dce820/cloudmapper/prepare.py#L307

Then it should find all the connections to a node within a VPC, and across the peering connections as shown here: https://github.com/duo-labs/cloudmapper/blob/a116da65b72af48398541eb2e37c768c76dce820/cloudmapper/prepare.py#L190

CloudMapper does not currently take routing tables into account.

There is likely a bug in the code as that VPC appears isolated.  I would first ensure that CloudMapper is figuring out the VPC peering connections in the `prepare` phase by adding some `print` statements here: https://github.com/duo-labs/cloudmapper/blob/a116da65b72af48398541eb2e37c768c76dce820/cloudmapper/prepare.py#L322

Add:
```
print "*** VPC peering found between {} and {}".format(accepter_id, requester_id)
```

Assuming CloudMapper knows about these peerings, then you would need to debug if CloudMapper is iterating through the nodes in the other VPC.  Please let me know what you find so I can fix this.

**Comment by vr00n** [comment]
Thanks for that thorough explanation @0xdabbad00 . 

I did add that piece of code in `prepare.py` and it did report 3 out of 4 peering connections that exist in that account. 

We have 4 peering connections in total in this account.

3 of them connect 1, 2, & 3 VPCs together (1->2, 2->3, and 1->3). 

The 4th peering connection connects resources from another account to 1-VPC (_that one that appears disconnected in the diagram_).

From a security group perspective there exists a self-referencing group to allow for the cross account connection.

**Comment by 0xdabbad00** [comment]
Currently, CloudMapper does not support cross-account peerings or really anything involving multiple accounts so it sounds like that explains this situation.  If so, I think we should close this ticket and create a ticket to add multiple account support as a feature (although it likely will be a while before I'm able to get to that).

**Comment by vr00n** [comment]
The VPC in question (`1-vpc` in the diagram above) still has peering connections to 2 and 3. It is also connected to a VPC in another account. So a connection should be shown between `1-vpc` to `2-vpc` AND `1-vpc` and `3-vpc`

**Comment by 0xdabbad00** [comment]
That shouldn't effect it.  I'll need to set up a similar environment or see the files `describe-vpc-peering-connections.json` and `describe-vpcs.json` to try to diagnose what is going on.

**Comment by 0xdabbad00** [comment]
Closing due to staleness.

**Comment by Narayan-PrasadK** [comment]
Thanks in advance. 
I have configured config.json and added cidrs for external cirds , when I check my report it is not showing VPC Peering in network visualizer, I'm I missing anything or is this any issue? I'm able to see ec2-describe-vpc-peering-connections.json file in account-data and in that file it contains only 
{
    "VpcPeeringConnections": []
}
~

Is something is missing? Do I need to do anything to collect data? 
-------------------------------------

And I have found some resource which is not using, so deleted it and again run collect, prepare for my account and checked in network visualizer. still I'm able to see that deleted resource. Do we need to do anything else? Is there any cache concept?
